plugin.program.AML/resources/disk_IO.py

```diff
@@ -0,0 +1,1308 @@
+# -*- coding: utf-8 -*-
+# Advanced MAME Launcher filesystem I/O functions
+#
+
+# Copyright (c) 2016-2018 Wintermute0110 <wintermute0110@gmail.com>
+#
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation; version 2 of the License.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+
+# --- Python standard library ---
+from __future__ import unicode_literals
+from __future__ import division
+import json
+import io
+import codecs
+import time
+import subprocess
+import re
+import threading
+import copy
+# import gc
+# import resource # Module not available on Windows
+
+# --- XML stuff ---
+# >> cElementTree sometimes fails to parse XML in Kodi's Python interpreter... I don't know why
+# import xml.etree.cElementTree as ET
+
+# >> Using ElementTree seems to solve the problem
+import xml.etree.ElementTree as ET
+
+# --- AEL packages ---
+from constants import *
```

Comment:
I know, but it saves a lot of typing. AML Python files are designed to be like C includes rather than Python modules. I took a lot of care to design a good hierarchy of the imports so there are no circular dependencies or redundancies and so wildcards can be used. Also, I stick to the convention to use _ functions that do not need to be exported by default.